src/wp-includes/rest-api/endpoints/class-wp-rest-autosaves-controller.php

```diff
@@ -390,11 +407,26 @@ public function create_post_autosave( $post_data ) {
 			do_action( 'wp_creating_autosave', $new_autosave );
 
 			// wp_update_post() expects escaped array.
-			return wp_update_post( wp_slash( $new_autosave ) );
+			$revision_id = wp_update_post( wp_slash( $new_autosave ) );
+		} else {
+			// Create the new autosave as a special post revision.
+			$revision_id = _wp_put_post_revision( $post_data, true );
+		}
+
+		if( is_wp_error( $revision_id ) ) {
```

Comment:
@spacedmonkey thinking we should check for 0 here as well since `_wp_put_post_revision` can return 0.